buildenv/jenkins/build

```diff
@@ -0,0 +1,248 @@
+                                    echo "error: Cannot specify 2 OMR Repos"
+                                    error()
+                                }
+                                break
+                            case "openj9-openjdk-jdk9":
+                                if (!openjdk_bool) {
+                                    openjdk_bool = true
+                                    OJDK_PR="${PR_ID}"
+                                    echo "Dependent OopenJDK change found: #${PR_ID}"
+                                } else {
+                                    echo "error: Cannot specify 2 OpenJDK Repos"
+                                    error()
+                                }
+                                break
+                            default:
+                                echo "error: Bad Depends Repo: '${REPO}' in '${DEPEND}'"
+                                error()
+                        }
+                    }
+                }
+                if (openjdk_bool) {
+                    sh "git fetch --tags --progress origin +refs/pull/*:refs/remotes/origin/pr/*"
+                    sh "git checkout refs/remotes/origin/pr/${OJDK_PR}/merge"
+                }
+                sh 'bash ./get_source.sh'
+                dir ('openj9') {
+                    sh "git fetch --tags --progress origin +refs/pull/*:refs/remotes/origin/pr/*"
+                    sh "git checkout ${sha1}"
+                }
+                if (omr_bool) {
+                    dir ('omr') {
+                        if (omr_upstream) {
+                            sh "git config remote.origin.url https://github.com/eclipse/omr.git"
+                        }
+                        sh "git fetch --tags --progress origin +refs/pull/*:refs/remotes/origin/pr/*"
+                        sh "git checkout refs/remotes/origin/pr/${OMR_PR}/merge"
+                    }
+                }
+            }
+        }
+    }
+}
+
+def build() {
+    stage('Compile') {
+        timestamps {
+            sh "bash ./configure --with-freemarker-jar=$FREEMARKER --with-boot-jdk=$BOOT_JDK"
+            sh "make all"
+        }
+    }
+    stage('Java Version') {
+        timestamps {
+            sh "build/$RELEASE/images/${(SDK_VERSION == '8') ? 'j2sdk-image' : 'jdk'}/bin/java -version"
+        }
+    }
+}
+
+def archive() {
+    stage('Archive') {
+        timestamps {
+            sh "tar -zcvf ${WORKSPACE}/${SDK_PREFIX}`date +%Y%d%m%H%M`${SDK_SUFFIX} build/$RELEASE/images/${JDK_FOLDER}"
+            sh "tar -zcvf ${WORKSPACE}/${TEST_PREFIX}`git rev-parse --short HEAD`${TEST_SUFFIX} openj9/test/"
+            archiveArtifacts artifacts: "**/${SDK_PREFIX}*${SDK_SUFFIX}, **/${TEST_PREFIX}*${TEST_SUFFIX}", fingerprint: true, onlyIfSuccessful: true
+        }
+    }
+}
+
+def build_all() {
+    get_source()
+    build()
+    archive()
+    git_clean()
+}
+
+def build_pr() {
+    get_source()
+    checkout_pullrequest()
+    build()
```

Comment:
Is this missing a `git_clean()`?

Reply:
No, the PR jobs that call this function will then do testing followed by a `git_clean()`.
I will add some comments to this code